pkg/handler/data_recorder_kinesis.go

```diff
@@ -0,0 +1,87 @@
+package handler
+
+import (
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/pkg/util"
+	"github.com/checkr/flagr/swagger_gen/models"
+
+	"github.com/a8m/kinesis-producer"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/sirupsen/logrus"
+)
+
+type kinesisRecorder struct {
+	enabled  bool
+	producer *producer.Producer
+}
+
+// NewKinesisRecorder creates a new Kinesis recorder
+var NewKinesisRecorder = func() DataRecorder {
+	client := kinesis.New(session.New(aws.NewConfig()))
+
+	p := producer.New(&producer.Config{
+		StreamName:          config.Config.RecorderKinesisStreamName,
+		Client:              client,
+		BacklogCount:        config.Config.RecorderKinesisBacklogCount,
+		MaxConnections:      config.Config.RecorderKinesisMaxConnections,
+		FlushInterval:       config.Config.RecorderKinesisFlushInterval,
+		BatchSize:           config.Config.RecorderKinesisBatchSize,
+		BatchCount:          config.Config.RecorderKinesisBatchCount,
+		AggregateBatchCount: config.Config.RecorderKinesisAggregateBatchCount,
+		AggregateBatchSize:  config.Config.RecorderKinesisAggregateBatchSize,
+		Verbose:             config.Config.RecorderKinesisVerbose,
+		Logger:              logrus.WithField("producer", "kinesis"),
+	})
+
+	p.Start()
+
+	go func() {
+		for err := range p.NotifyFailures() {
+			logrus.WithField("kinesis_error", err).Error("error pushing")
```

Comment:
nit, the error message can be more specific, like `logrus.WithField("kinesis_error", err).Error("error pushing to kinesis")`, same for the error messages below

Reply:
👍